runtime/src/main/java/org/corfudb/runtime/LogReplicationServiceClient.java

```diff
@@ -0,0 +1,339 @@
+
+    private final CorfuStore corfuStore;
+    private final Table<ClientRegistrationId, ClientRegistrationInfo, Message> replicationRegistrationTable;
+    private final Table<ClientDestinationInfoKey, DestinationInfoVal, Message> sourceMetadataTable;
+
+    private final ClientRegistrationId clientKey;
+    private final ClientRegistrationInfo clientInfo;
+
+    private final String clientName;
+    private final String modelEnum;
+
+    /**
+     * Constructor for LogReplicationClient
+     *
+     * @param runtime Corfu Runtime
+     * @param clientName String representation of the client name
+     * @throws InvocationTargetException InvocationTargetException
+     * @throws NoSuchMethodException NoSuchMethodException
+     * @throws IllegalAccessException IllegalAccessException
+     */
+    public LogReplicationServiceClient(CorfuRuntime runtime, String clientName) throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+        this.corfuStore = new CorfuStore(runtime);
+        this.clientName = clientName;
+
+        this.modelEnum = "LOGICAL_GROUPS";
+
+        this.clientKey = ClientRegistrationId.newBuilder()
+                .setClientName(clientName)
+                .build();
+        this.clientInfo = ClientRegistrationInfo.newBuilder()
+                .setClientName(clientName)
+                .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                .setRegistrationTime(Timestamp.newBuilder().setSeconds(new Date().getTime() / 1000).build())
+                .build();
+
+        this.replicationRegistrationTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LR_REGISTRY_TABLE_NAME, ClientRegistrationId.class,
+                ClientRegistrationInfo.class, null, TableOptions.builder().build());
+        this.sourceMetadataTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LR_SOURCE_METADATA_TABLE_NAME, ClientDestinationInfoKey.class,
+                DestinationInfoVal.class, null, TableOptions.builder().build());
+
+        register();
+    }
+
+    /**
+     * Adds client and model to LogReplicationRegistrationTable
+     */
+    private void register() {
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    if (txn.isExists(LR_REGISTRY_TABLE_NAME, clientKey)) {
+                        log.warn("Client already registered \nclientKey: {} \nclientInfo: {}",
+                                clientKey, txn.getRecord(replicationRegistrationTable, clientKey).getPayload().toString());
+                    }
+                    else {
+                        txn.putRecord(replicationRegistrationTable, clientKey, clientInfo, null);
+                        txn.commit();
+                    }
+                }
+                catch (Exception e) {
+                    log.error("Unable to register client {}, Error: {}", clientName, e);
+                }
```

Comment:
Actually the retry here won't automatically work, unless you throw `RetryNeededException` somewhere in your lambda function, as internally it catches that exception to trigger next run:
https://github.com/CorfuDB/CorfuDB/blob/dcab1d42738ba183bdd04a5feffd6a3b9dad4594/runtime/src/main/java/org/corfudb/util/retry/IRetry.java#L74-L83
take this as an example:
https://github.com/CorfuDB/CorfuDB/blob/dcab1d42738ba183bdd04a5feffd6a3b9dad4594/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java#L521